src/parser.cpp

```diff
@@ -11,6 +11,22 @@ namespace vpd
 Parser::Parser(const std::string& vpdFilePath, nlohmann::json parsedJson) :
     m_vpdFilePath(vpdFilePath), m_parsedJson(parsedJson)
 {
+    std::error_code l_errCode;
+    auto l_filesize = -1;
+
+    if (!std::filesystem::exists(m_vpdFilePath, l_errCode) ||
+        ((l_filesize = std::filesystem::file_size(m_vpdFilePath, l_errCode)) ==
+         0))
+    {
+        std::string l_message =
```

Comment:
can be `const std::string l_message` . Also, I think its better to use uniform initialization for variables (using {}).

Reply:
done